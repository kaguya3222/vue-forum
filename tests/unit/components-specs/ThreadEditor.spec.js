import { mount } from "@vue/test-utils";
import ThreadEditor from "../../../src/components/ThreadEditor";

describe("ThreadEditor", () => {
  const wrapper = mount(ThreadEditor);
  wrapper.vm.$emit = jest.fn();
  test("Shows thread edition form", () => {
    expect(wrapper).toMatchSnapshot();
  });
  describe("save method", () => {
    beforeEach(() => {
      const threadEditionForm = wrapper.find("form");
      threadEditionForm.trigger("submit");
    });
    const saveMethodSpy = jest.spyOn(wrapper.vm, "save");
    test("Is called when form is submit", () => {
      expect(saveMethodSpy).toHaveBeenCalled();
    });
    test("Emits 'save' event", () => {
      expect(wrapper.vm.$emit).toHaveBeenCalledWith("save", {
        text: wrapper.vm.form.text,
        title: wrapper.vm.form.title
      });
    });
  });
  describe("cancel method", () => {
    beforeEach(() => {
      const cancelButton = wrapper.find(".btn-ghost");
      cancelButton.trigger("click");
    });
    const cancelMethodSpy = jest.spyOn(wrapper.vm, "cancel");
    test("Is called when cancel button is clicked", () => {
      expect(cancelMethodSpy).toHaveBeenCalled();
    });
    test("Emits 'cancel' event", () => {
      expect(wrapper.vm.$emit).toHaveBeenCalledWith("cancel");
    });
  });
});
