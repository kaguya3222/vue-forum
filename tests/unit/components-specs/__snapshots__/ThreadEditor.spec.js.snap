// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadEditor Shows thread edition form 1`] = `
<form>
  <div class="form-group"><label for="thread_title">Title:</label> <input type="text" id="thread_title" name="title" class="form-input"></div>
  <div class="form-group"><label for="thread_content">Content:</label> <textarea id="thread_content" name="content" rows="8" cols="140" class="form-input"></textarea></div>
  <div class="btn-group"><button class="btn btn-ghost">
      Cancel
    </button> <button type="submit" name="Publish" class="btn btn-blue">
      Publish
    </button></div>
</form>
`;
